cookbook/src/component.md

```diff
@@ -1,29 +1,65 @@
 # Components
-A [Component][component] is a fold function (logic), a state variable and a [ViewBuilder][builder]
-all wrapped up in one, for your convenience!
-
-## Creating a Component
-A mogwai component can be created by implementing the [Component][component]
-trait for any type. That type is the state. Its [Component::update][update] function
-is the logic. The [Component::view][view] function returns a builder that becomes the view
-(or views). There are a couple steps to set up the model-view-controller scenario:
-
-  1. Use [Gizmo::from][gizmo_from] to turn your state type into a [Gizmo][gizmo]
-     `let gizmo = Gizmo::from(my_data);`
-  2. Create a view using a builder from the gizmo
-     `let view = View::from(gizmo.view_builder());`
-  3. Run the view and communicate with it using the gizmo
-     ```rust,ignore
-     view.run().unwrap_throw();
-     gizmo.send(&MyMessage);
-     ```
-
-Alternatively, if you don't have a need to communicate with your view you can create a view
-directly from the gizmo with `let view = View::from(gizmo);`.
+The [Component][traitcomponent] trait allows rust types to be used as a node
+in the DOM. These nodes are loosely referred to as components.
+
+## Discussion
+A type that implements [Component][traitcomponent] is the state of a user
+interface element - for example we might use the following struct to express a
+component that keeps track of how many clicks a user has performed on a button.
+```rust
+struct ButtonCounter {
+    clicks: u32
+}
+```
+or `current_contents: String` to store what a
+user has entered into an input field. Another common pattern is to use a field like
+`items: Gizmo<String>` to be able to communicate with a list of subcopmonents (we'll go
+into more detail on that in the [subcomponents](nest_component.md) chapter).
+
+There are two basic concepts you must understand about components:
+
+1. **State**
+   As described above, a component probably contains some stateful information which
+   determines how that component reacts to input messages (also known as your component's
+   [Self::ModelMsg][traitcomponent_atypemodelmsg] type).
+   In turn this determines when and if the component sends output messages (also known as
+   your component's [Self::ViewMsg][traitcomponent_atypeviewmsg]) to its view.
+2. **View**
+   A component describes the structure of its DOM, most likely using [RSX](rsx.md).
+   It also describes how messages received may change that view and when or if the view
+   sends input messages to update the component's state.
```

Comment:
I think it would be helpful to note that the view is only run one time (at least this was how it started to click for me). For folks coming from other libraries (ahem, React) this looks very much like the `render` function but that's not how mogwai works. The `view` function is like a "setup" function that describes initial state (the RSX) and how it will update (the `Receiver`s).

Reply:
Great point!